lua/weapons/weapon_Auto_Shotgun.lua

```diff
@@ -0,0 +1,135 @@
+-- Easy Settings part
+
+-- SWEP Info
+SWEP.Author = "yousifgaming"
+SWEP.Instructions = "Primary is the same as Secondary fire automatic shotgun"
+SWEP.Purpose = "To play with Dingus"
+SWEP.Contact = "Don't Contact me i dont like people"
+
+-- SWEP info in spawnmenu
+SWEP.PrintName = "Auto Shotgun"
+SWEP.Category = "yousifgaming's Weapons"
+SWEP.Spawnable = true
+SWEP.AdminOnly = false
+SWEP.IconOverride = "materials/cat.jpg"
+
+if CLIENT then
+    SWEP.WepSelectIcon = surface.GetTextureID('cat')
+end
+
+-- Primary settings
+SWEP.Primary.ClipSize = 30
+SWEP.Primary.DefaultClip = 30
+SWEP.Primary.Automatic = true
+SWEP.Primary.Ammo = "Buckshot"
+
+-- Secondary settings
+SWEP.Secondary.Ammo = "None"
+SWEP.Secondary.Automatic = true
+
+-- Reload status
+SWEP.ReloadSound = "weapons/shotgun/shotgun_reload" .. math.random(1, 3) .. ".wav"
+SWEP.ReloadInProgress = false
+
+-- Switch Pool
+SWEP.Weight = 5
+SWEP.AutoSwitchTo = false
+SWEP.AutoSwitchFrom = false
+
+-- Slot location and gui elements
+SWEP.Slot = 1
+SWEP.SlotPos = 0
+SWEP.DrawAmmo = true
+SWEP.DrawCrosshair = true
+
+-- World and View Models
+SWEP.ViewModel = "models/weapons/c_shotgun.mdl"
+SWEP.WorldModel = "models/weapons/w_shotgun.mdl"
+function SWEP:Initialize()
+    self:SetHoldType("shotgun")
+end
+
+-- Uses hands?
+SWEP.UseHands = true
+
+-- Shooting sound
+SWEP.ShootSound = Sound("weapons/xm1014/xm1014-1.wav")
+
+-- Hard Parts / Functions
+
+-- Primary Attack Function

```

Comment:
nit: I also don't think these comments help with code documentation, you're just commenting what the code does than why.